### DISCUSSION TITLE
Running the application gives an error that says "ComputeIsotropicElasticityTensor" is not a registered object even though I have enabled all the modules in the makefile.

### DISCUSSION THREAD

**Discussion by GraphicsMonster** [idaholab/moose#26517] [Q&A Modules: General] [answered]
Pretty much what the title says. I encountered this error and noticed that I had not enabled the makefile to have all the modules enabled so I quickly enabled them all and ran "make" again. It took a long time but went with no errors. I tried running my input file again and it still shows the same error.

All help will be super appreciated!
![moose_app_error](https://github.com/idaholab/moose/assets/63590510/136397dc-7023-4b13-9084-d68d01d04880)
![Screenshot from 2024-01-09 23-24-44](https://github.com/idaholab/moose/assets/63590510/8c835b90-d9df-4ba6-8054-836524ea84aa)

**Comment by GiudGiud** [comment]
Hello

Can you please run ldd on the executable? If you are on Linux: `ldd <executable-opt>`
On Mac it s `otool -L <executable-opt>`

I want to check that the modules are indeed linked to the application. It can take a recompile to fix this if not

Guillaume

**Reply by GraphicsMonster** [reply]
Super sorry for the late response. I am travelling.
Here I ran ldd:
`moose) devansh@devansh-VirtualBox:~/Desktop/moose-apps/thermal_sim$ ldd ./thermal_sim-opt

	linux-vdso.so.1 (0x00007fff9ffc6000)

	libthermal_sim_test-opt.so.0 => /home/devansh/Desktop/moose-apps/thermal_sim/test/lib/libthermal_sim_test-opt.so.0 (0x00007fa151d8b000)

	libthermal_sim-opt.so.0 => /home/devansh/Desktop/moose-apps/thermal_sim/lib/libthermal_sim-opt.so.0 (0x00007fa151d65000)

	libmodule_loader_with_cr_em_eps_fp_fet_gc_ls_ray_rdg_rct_rich_st_ht_sp_tm_ns_con_fsi_misc_opt_pd_pf_pflow_th_st_xfem_comb-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/module_loader/lib/libmodule_loader_with_cr_em_eps_fp_fet_gc_ls_ray_rdg_rct_rich_st_ht_sp_tm_ns_con_fsi_misc_opt_pd_pf_pflow_th_st_xfem_comb-opt.so.0 (0x00007fa151d52000)

	libcombined-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/combined/lib/libcombined-opt.so.0 (0x00007fa151d42000)

	libxfem-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/xfem/lib/libxfem-opt.so.0 (0x00007fa151a00000)

	libscalar_transport-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/scalar_transport/lib/libscalar_transport-opt.so.0 (0x00007fa151cd0000)

	libthermal_hydraulics-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/thermal_hydraulics/lib/libthermal_hydraulics-opt.so.0 (0x00007fa151000000)

	libporous_flow-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/porous_flow/lib/libporous_flow-opt.so.0 (0x00007fa150a00000)

	libphase_field-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/phase_field/lib/libphase_field-opt.so.0 (0x00007fa150000000)

	libperidynamics-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/peridynamics/lib/libperidynamics-opt.so.0 (0x00007fa14fc00000)

	liboptimization-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/optimization/lib/liboptimization-opt.so.0 (0x00007fa150863000)

	libmisc-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/misc/lib/libmisc-opt.so.0 (0x00007fa151977000)

	libfsi-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/fsi/lib/libfsi-opt.so.0 (0x00007fa151905000)

	libcontact-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/contact/lib/libcontact-opt.so.0 (0x00007fa14f800000)

	libnavier_stokes-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/navier_stokes/lib/libnavier_stokes-opt.so.0 (0x00007fa14ee00000)

	libtensor_mechanics-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.so.0 (0x00007fa14e000000)

	libsolid_properties-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/solid_properties/lib/libsolid_properties-opt.so.0 (0x00007fa14ff50000)

	libheat_transfer-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/heat_transfer/lib/libheat_transfer-opt.so.0 (0x00007fa14dc00000)

	libstochastic_tools-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/stochastic_tools/lib/libstochastic_tools-opt.so.0 (0x00007fa14d400000)

	librichards-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/richards/lib/librichards-opt.so.0 (0x00007fa14d000000)

	libreactor-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/reactor/lib/libreactor-opt.so.0 (0x00007fa14fa5b000)

	librdg-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/rdg/lib/librdg-opt.so.0 (0x00007fa14fe9f000)

	libray_tracing-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/ray_tracing/lib/libray_tracing-opt.so.0 (0x00007fa14ce07000)

	liblevel_set-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/level_set/lib/liblevel_set-opt.so.0 (0x00007fa14f73a000)

	libgeochemistry-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/geochemistry/lib/libgeochemistry-opt.so.0 (0x00007fa14cc55000)

	libfunctional_expansion_tools-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/functional_expansion_tools/lib/libfunctional_expansion_tools-opt.so.0 (0x00007fa14db39000)

	libfluid_properties-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/fluid_properties/lib/libfluid_properties-opt.so.0 (0x00007fa14c800000)

	libexternal_petsc_solver-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/external_petsc_solver/lib/libexternal_petsc_solver-opt.so.0 (0x00007fa1518cb000)

	libelectromagnetics-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/electromagnetics/lib/libelectromagnetics-opt.so.0 (0x00007fa14d32f000)

	libchemical_reactions-opt.so.0 => /home/devansh/Desktop/moose-project/moose/modules/chemical_reactions/lib/libchemical_reactions-opt.so.0 (0x00007fa14cb3e000)

	libmoose-opt.so.0 => /home/devansh/Desktop/moose-project/moose/framework/libmoose-opt.so.0 (0x00007fa14a000000)

	libpng16.so.16 => /home/devansh/mambaforge3/envs/moose/lib/libpng16.so.16 (0x00007fa15188f000)

	libpcre-opt.so.0 => /home/devansh/Desktop/moose-project/moose/framework/contrib/pcre/libpcre-opt.so.0 (0x00007fa150fd3000)

	libhit-opt.so.0 => /home/devansh/Desktop/moose-project/moose/framework/contrib/hit/libhit-opt.so.0 (0x00007fa14fe31000)

	libwaspcore.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwaspcore.so.04 (0x00007fa150fb2000)

	libwaspddi.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwaspddi.so.04 (0x00007fa150844000)

	libwaspexpr.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwaspexpr.so.04 (0x00007fa14fa27000)

	libwasphalite.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwasphalite.so.04 (0x00007fa14edab000)

	libwasphit.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwasphit.so.04 (0x00007fa14dfb2000)

	libwasphive.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwasphive.so.04 (0x00007fa151cb8000)

	libwaspjson.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwaspjson.so.04 (0x00007fa14daf5000)

	libwasplsp.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwasplsp.so.04 (0x00007fa14dab8000)

	libwaspsiren.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwaspsiren.so.04 (0x00007fa14fa0b000)

	libwaspson.so.04 => /home/devansh/mambaforge3/envs/moose/wasp/lib/libwaspson.so.04 (0x00007fa14d2e1000)

	libmesh_opt.so.0 => /home/devansh/mambaforge3/envs/moose/libmesh/lib/libmesh_opt.so.0 (0x00007fa148800000)

	libtimpi_opt.so.11 => /home/devansh/mambaforge3/envs/moose/libmesh/lib/libtimpi_opt.so.11 (0x00007fa14fe22000)

	libhdf5_cpp.so.200 => /home/devansh/mambaforge3/envs/moose/lib/libhdf5_cpp.so.200 (0x00007fa14d269000)

	libvtkIOCore-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkIOCore-9.2.so.1 (0x00007fa14cad0000)

	libvtkCommonCore-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkCommonCore-9.2.so.1 (0x00007fa147e00000)

	libvtkCommonDataModel-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkCommonDataModel-9.2.so.1 (0x00007fa147800000)

	libvtkFiltersCore-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkFiltersCore-9.2.so.1 (0x00007fa146e00000)

	libvtkIOXML-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkIOXML-9.2.so.1 (0x00007fa149ee6000)

	libvtkImagingCore-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkImagingCore-9.2.so.1 (0x00007fa146a00000)

	libvtkIOImage-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkIOImage-9.2.so.1 (0x00007fa146822000)

	libvtkImagingMath-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkImagingMath-9.2.so.1 (0x00007fa14c7b7000)

	libvtkIOParallelXML-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkIOParallelXML-9.2.so.1 (0x00007fa14c76e000)

	libvtkParallelMPI-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkParallelMPI-9.2.so.1 (0x00007fa14f71e000)

	libvtkParallelCore-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkParallelCore-9.2.so.1 (0x00007fa14c70d000)

	libvtkCommonExecutionModel-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtkCommonExecutionModel-9.2.so.1 (0x00007fa148702000)

	libvtksys-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/libvtksys-9.2.so.1 (0x00007fa149e95000)

	libz.so.1 => /home/devansh/mambaforge3/envs/moose/lib/libz.so.1 (0x00007fa14ed90000)

	libslepc.so.3.019 => /home/devansh/mambaforge3/envs/moose/lib/libslepc.so.3.019 (0x00007fa146400000)

	libpetsc.so.3.20 => /home/devansh/mambaforge3/envs/moose/lib/libpetsc.so.3.20 (0x00007fa144800000)

	libHYPRE-2.29.0.so => /home/devansh/mambaforge3/envs/moose/lib/libHYPRE-2.29.0.so (0x00007fa144200000)

	libstrumpack.so.7.1 => /home/devansh/mambaforge3/envs/moose/lib/libstrumpack.so.7.1 (0x00007fa143000000)

	libscalapack.so.2.2 => /home/devansh/mambaforge3/envs/moose/lib/libscalapack.so.2.2 (0x00007fa142800000)

	libsuperlu_dist.so.8 => /home/devansh/mambaforge3/envs/moose/lib/libsuperlu_dist.so.8 (0x00007fa14767c000)

	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa151888000)

	libparmetis.so => /home/devansh/mambaforge3/envs/moose/lib/libparmetis.so (0x00007fa149e57000)

	libmetis.so => /home/devansh/mambaforge3/envs/moose/lib/libmetis.so (0x00007fa148697000)

	libhdf5_hl.so.200 => /home/devansh/mambaforge3/envs/moose/lib/libhdf5_hl.so.200 (0x00007fa14caac000)

	libhdf5.so.200 => /home/devansh/mambaforge3/envs/moose/lib/libhdf5.so.200 (0x00007fa142200000)

	libmpifort.so.12 => /home/devansh/mambaforge3/envs/moose/lib/libmpifort.so.12 (0x00007fa147daf000)

	libgfortran.so.5 => /home/devansh/mambaforge3/envs/moose/lib/libgfortran.so.5 (0x00007fa144055000)

	libgcc_s.so.1 => /home/devansh/mambaforge3/envs/moose/lib/libgcc_s.so.1 (0x00007fa14ed75000)

	libquadmath.so.0 => /home/devansh/mambaforge3/envs/moose/lib/libquadmath.so.0 (0x00007fa149e1e000)

	libstdc++.so.6 => /home/devansh/mambaforge3/envs/moose/lib/libstdc++.so.6 (0x00007fa14201d000)

	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa150fad000)

	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa15083f000)

	libmpi.so.12 => /home/devansh/mambaforge3/envs/moose/lib/libmpi.so.12 (0x00007fa141c00000)

	libmpicxx.so.12 => /home/devansh/mambaforge3/envs/moose/lib/libmpicxx.so.12 (0x00007fa148678000)

	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa146d19000)

	libgomp.so.1 => /home/devansh/mambaforge3/envs/moose/lib/libgomp.so.1 (0x00007fa147d73000)

	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa141800000)

	libnetcdf.so.13 => /home/devansh/mambaforge3/envs/moose/libmesh/lib/./libnetcdf.so.13 (0x00007fa1462f0000)

	libcrypto.so.1.1 => /home/devansh/mambaforge3/envs/moose/lib/./libcrypto.so.1.1 (0x00007fa141400000)

	libcurl.so.4 => /home/devansh/mambaforge3/envs/moose/lib/./libcurl.so.4 (0x00007fa14475d000)

	libvtkdoubleconversion-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkdoubleconversion-9.2.so.1 (0x00007fa14dfa1000)

	libvtklz4-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtklz4-9.2.so.1 (0x00007fa147d3f000)

	libvtklzma-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtklzma-9.2.so.1 (0x00007fa146cec000)

	libvtkzlib-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkzlib-9.2.so.1 (0x00007fa14765e000)

	libvtkloguru-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkloguru-9.2.so.1 (0x00007fa146cb4000)

	/lib64/ld-linux-x86-64.so.2 (0x00007fa151da9000)

	libvtkCommonTransforms-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkCommonTransforms-9.2.so.1 (0x00007fa1467df000)

	libvtkCommonMisc-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkCommonMisc-9.2.so.1 (0x00007fa140e00000)

	libvtkCommonSystem-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkCommonSystem-9.2.so.1 (0x00007fa1462b0000)

	libvtkpugixml-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkpugixml-9.2.so.1 (0x00007fa144720000)

	libvtkCommonMath-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkCommonMath-9.2.so.1 (0x00007fa146289000)

	libvtkIOXMLParser-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkIOXMLParser-9.2.so.1 (0x00007fa14c6f3000)

	libvtkDICOMParser-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkDICOMParser-9.2.so.1 (0x00007fa146c97000)

	libvtkmetaio-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkmetaio-9.2.so.1 (0x00007fa144682000)

	libvtkpng-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkpng-9.2.so.1 (0x00007fa142fb9000)

	libvtktiff-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtktiff-9.2.so.1 (0x00007fa142f37000)

	libvtkjpeg-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkjpeg-9.2.so.1 (0x00007fa142783000)

	libvtkIOLegacy-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/./libvtkIOLegacy-9.2.so.1 (0x00007fa1426c4000)

	libzfp.so.0 => /home/devansh/mambaforge3/envs/moose/lib/./libzfp.so.0 (0x00007fa142670000)

	libnghttp2.so.14 => /home/devansh/mambaforge3/envs/moose/lib/././libnghttp2.so.14 (0x00007fa144029000)

	libssh2.so.1 => /home/devansh/mambaforge3/envs/moose/lib/././libssh2.so.1 (0x00007fa141bbc000)

	libssl.so.1.1 => /home/devansh/mambaforge3/envs/moose/lib/././libssl.so.1.1 (0x00007fa141b2c000)

	libgssapi_krb5.so.2 => /home/devansh/mambaforge3/envs/moose/lib/././libgssapi_krb5.so.2 (0x00007fa141ad9000)

	libvtkkissfft-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/././libvtkkissfft-9.2.so.1 (0x00007fa14f706000)

	libvtkexpat-9.2.so.1 => /home/devansh/mambaforge3/envs/moose/libmesh-vtk/lib/././libvtkexpat-9.2.so.1 (0x00007fa142f04000)

	libkrb5.so.3 => /home/devansh/mambaforge3/envs/moose/lib/./././libkrb5.so.3 (0x00007fa141729000)

	libk5crypto.so.3 => /home/devansh/mambaforge3/envs/moose/lib/./././libk5crypto.so.3 (0x00007fa1467c5000)

	libcom_err.so.3 => /home/devansh/mambaforge3/envs/moose/lib/./././libcom_err.so.3 (0x00007fa14dab2000)

	libkrb5support.so.0 => /home/devansh/mambaforge3/envs/moose/lib/./././libkrb5support.so.0 (0x00007fa149e0f000)

	libkeyutils.so.1 => /home/devansh/mambaforge3/envs/moose/lib/./././libkeyutils.so.1 (0x00007fa14caa5000)

	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fa14466e000)

`

**Reply by GiudGiud** [reply]
This looks correct.
Did you `make cleanall` after adding the new modules to your app?

**Reply by GraphicsMonster** [reply]
No just ran `make`
Should I do that?

**Reply by GiudGiud** [reply]
yup please. Either make cleanall
or save all your work then `git clean -xfd` then rebuild

this will take a while but I dont see the cause

**Reply by GraphicsMonster** [reply]
Right. I'll report back very soon(still travelling) with the results.

**Reply by GraphicsMonster** [reply]
I ran make cleanall which essentially deleted the already present executable from my application and then ran make again(which took like 2 hours). Now even after running  make and it having compiled everything successfully, no executable has been added to the working directory and when i try to run ./run_tests it fails.
![Screenshot from 2024-01-11 13-30-29](https://github.com/idaholab/moose/assets/63590510/6254c139-184f-46e0-8a90-713d7ce64a83)

**Reply by GiudGiud** [reply, marked as answer]
The build failed. 

Can you go build the combined executable in moose/modules and see if that succeeds?

Then if it does, can you make clean in your app (it should not remove the built moose modules, just your app) and make there

**Reply by GraphicsMonster** [reply]
so you want me to go to moose/modules and run make there? and then if it succeeds run make clean in my application's directory?

**Reply by GiudGiud** [reply]
yes

**Reply by GraphicsMonster** [reply]
That worked!
Thank you so much. I love you guys!
